web/src/app/alerts/components/AlertDetails.js

```diff
@@ -27,15 +28,153 @@ import Options from '../../util/Options'
 import gql from 'graphql-tag'
 import PageActions from '../../util/PageActions'
 import Markdown from '../../util/Markdown'
+import { graphql } from 'react-apollo'
 
 const localStorage = window.localStorage
 const exactTimesKey = 'show_exact_times'
 
-const sortTime = (a, b) => {
-  const ma = moment(a.timestamp)
-  const mb = moment(b.timestamp)
-  if (ma.isSame(mb)) return 0
-  return ma.isAfter(mb) ? -1 : 1
+const LIMIT = 149
+
+const query = gql`
+  query getAlert($id: Int!, $input: AlertRecentEventsOptions) {
+    alert(id: $id) {
+      data: recentEvents(input: $input) {
+        nodes {
+          timestamp
+          message
+        }
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+      }
+    }
+  }
+`
+
+@graphql(query, {
```

Comment:
It might be worth bringing this component out into its own file and using the hooks API. It's a lot more straightforward than previous versions of Apollo

https://www.apollographql.com/docs/tutorial/queries/#build-a-paginated-list